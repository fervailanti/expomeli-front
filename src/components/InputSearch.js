import React, { useState } from 'react'
import { images } from '../images'

const InputSearch = ({ onSearch }) => {
  const [query, setSearch] = useState('')

  const onInputChange = (e) => setSearch(e.target.value)

  const onSubmit = (e) => {
    onSearch(query)
    e.preventDefault()
  }

  return (
    <form onSubmit={onSubmit} className="search-box">
      <input
        className="search-input"
        type="text"
        value={query}
        onChange={onInputChange}
        placeholder="Buscar productos, marcas y mÃ¡s..."
      />
      <button type="submit" className="search-btn">
        <img src={images.searchIcon} className="search-icon" alt="search_icon" />
      </button>
    </form>
  )
}

export default InputSearch
