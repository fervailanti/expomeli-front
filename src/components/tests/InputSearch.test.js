import React from 'react'
import InputSearch from '../InputSearch'
import { render } from '@testing-library/react'

it('InputSearch component should render correctly', () => {
  const tree = render(<InputSearch onSearch={(e) => e} />)
  expect(tree.container).toMatchSnapshot()
})
