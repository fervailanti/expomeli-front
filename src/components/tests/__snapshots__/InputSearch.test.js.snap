// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch component should render correctly 1`] = `
<div>
  <form
    class="search-box"
  >
    <input
      class="search-input"
      placeholder="Buscar productos, marcas y mÃ¡s..."
      type="text"
      value=""
    />
    <button
      class="search-btn"
      type="submit"
    >
      <img
        alt="search_icon"
        class="search-icon"
        src="[object Object]"
      />
    </button>
  </form>
</div>
`;
